### ISSUE TITLE
Performance issues (May be out of scope)

### ISSUE THREAD

**Issue by CanMehmetK** [grandnode/grandnode#997]
Category issue is same llike nopCommerce Id,Parent id nedted relation for this kind project is not preferable.

I only have 34000 categories and 2 Product for each category and system is down. CPU and RAM usage for mongo is killing computer.

I suggest limit sub category like Root,Sub1,Sub2,Sub3,Sub4,Sub5 and limit with this.
Becouse unlimited and unknown count of leaf tree is not an option.

By this Caching tree is more simple , calculating product count by category and with full subs counts may be is an option otherwise performance issue appering.

Performance in Admin is cınfusing also ! Must be more clear.
Suggestion ;
Also Shared class policy is fine but some of WebUI project calsses must move to Framework.

Not is an offencse. Just i think some other people will face with same issues ...now or in future.

**Comment by PatrykPorabik** [comment]
In fact, there is always room for maneuver to improve performance. However, it is very hard for me to imagine a situation where someone creates 34,000 categories nested within themselves. In the case of very large stores and databases, the process of store optimization begins at the stage of planning the product database, connections between them, nests create the most efficient scenario. 

But thanks for the suggestions, if we find a suitable and good way to improve performance, we will definitely implement it. The development of our application among others is based on performance improvements.

**Comment by Dredy** [comment]
I disagree with this statement "the process of store optimization begins at the stage of planning the product database".
Not always...
In my case (our Customer) we have 18 front categories, 18 distinct sectors with completely different products (household, electronics, clothing, wines, stationery, tools, hardware i.e.), every single Front category contains a tree of categories that separates groups of products coming from 12 external feeds (refreshed many times a day)
The website is a little Marketplace (a tiny-tiny-tiny-tiny amazon-like website) with 20.000 categories and 300.000 products
I think that 80% of the categories are very optimized and strict necessary for the correct positioning of the products.
Today we are using our custom system that is performing super fast (200ms average - category page server rendering with pagesize=50), but Your system is very better under usability, integrations and so on.
For the categories backoffice page Your system needs to collect the whole tree, that took 10 seconds per page, today we cannot switch to GrandNode, although it is really well done, due to some performance issues, but i will give a try to fork the project and find a better way to handle the category tree.
I did not try to import the whole products catalog to test the overall performance, maybe I will do it during the week-end.
